test/celeritas/ext/Vecgeom.test.cc

```diff
@@ -425,35 +413,157 @@ TEST_F(FourLevelsTest, TEST_IF_CELERITAS_CUDA(device))
-        test::GlobalGeoTestBase::reset_geometry();
+        return "solids";
+    }
+};
+
+//---------------------------------------------------------------------------//
+
+TEST_F(SolidsTest, accessors)
+{
+    if (starts_with(celeritas_vecgeom_version, "1.1"))
+    {
+        GTEST_SKIP() << "This geometry crashes when loading in VecGeom 1.1.17";
     }
 
-    void SetUp() override
+    const auto& geom = *this->geometry();
+    // TODO: this should be 31ish?
+    EXPECT_EQ(25, geom.num_volumes());
+    EXPECT_EQ(2, geom.max_depth());
+
+    EXPECT_EQ("World", geom.id_to_label(VolumeId{24}).name);
+    EXPECT_EQ("vol0", geom.id_to_label(VolumeId{4}).name);
+    EXPECT_EQ("vol1", geom.id_to_label(VolumeId{5}).name);
+    EXPECT_EQ("vol11", geom.id_to_label(VolumeId{9}).name);
+}
+
+//---------------------------------------------------------------------------//
+
+TEST_F(SolidsTest, DISABLED_trace)
+{
+    {
+        SCOPED_TRACE("Center +x");
+        auto result = this->track({-100, 0, 0}, {1, 0, 0});
+        result.print_expected();
+    }
+    {
+        SCOPED_TRACE("Upper +x");
+        auto result = this->track({-100, 12.5, 0}, {1, 0, 0});
+        result.print_expected();
+    }
+    {
+        SCOPED_TRACE("Lower +x");
+        auto result = this->track({-100, -12.5, 0}, {1, 0, 0});
+        result.print_expected();
+    }
+    ADD_FAILURE() << "These are all wrong...";
+}
+
+//---------------------------------------------------------------------------//
+// CONSTRUCT FROM GEANT4
+//---------------------------------------------------------------------------//
+
+class GeantBuilderTestBase : virtual public GeantTestBase,
+                             public VecgeomTestBase
+{
+  public:
+    static void SetUpTestCase()
     {
-        VecgeomTestBase::SetUp();
-        world_volume_
-            = load_gdml(this->test_data_path("celeritas", "four-levels.gdml"));
+        // Make sure existing VecGeom geometry has been cleared
+        GlobalGeoTestBase::reset_geometry();
     }
 
     SPConstGeo build_geometry() override
     {
-        CELER_NOT_IMPLEMENTED("build_geometry");
-        // return std::make_shared<VecgeomParams>(world_volume.get());
+        return std::make_shared<VecgeomParams>(this->get_world_volume());
     }
 
-  private:
-    UPG4PhysicalVolume world_volume_;
+    bool      enable_fluctuation() const final { return false; }
+    bool      enable_msc() const final { return false; }
+    bool      combined_brems() const final { return false; }
+    real_type secondary_stack_factor() const final { return 0; }
 };
+
+//---------------------------------------------------------------------------//
+
+#define SolidsGeantTest TEST_IF_CELERITAS_GEANT(SolidsGeantTest)
+class SolidsGeantTest : public GeantBuilderTestBase
+{
+  public:
+    const char* geometry_basename() const final { return "solids"; }
+};
+
+//---------------------------------------------------------------------------//
+
+TEST_F(SolidsGeantTest, accessors)
```

Comment:
Need a skip for this one too 😉

Reply:
Boo I had hoped it would be a bug in the importer for the original, not the re exported one. :(

Reply:
Yeah, unfortunately it crashes loading the Geant4-exported one too.